frontend/src/chooseAdventure/chooseAdventure.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import IntroView from './introView.js';
+import StageView from './stageView.js';
+import EndView from './endView.js';
+
+
+/**
+ * Component for displaying choose your own adventure skeleton
+ */
+export class ChooseAdventureView extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            view: 'intro',
+            history: [],
+            successTotal: 0,
+        };
+    }
+
+    componentDidMount() {
+        // do your fetch and set initial state
+    }
+
+    setView = (view) => {
+        this.setState({ view: view, });
+    };
+
+    updateHistory = (option) => {
+        this.setState((prevState) => ({
+            history: prevState.history.concat(option),
+        }));
+    };
+
+    updateSuccess = (successFactor) => {
+        this.setState((prevState) => ({
+            successTotal: prevState.successTotal + successFactor,
```

Comment:
This being additive feels weird to me -- I was expecting each step to have a likelihood of success, and that you'd get some kind of compounding probability thing at the end. When I play this otherwise the "success score" doesn't feel meaningful -- whereas some kind of probably would.